forem/forem
# [Search 2.0] Comments on search page

## What type of PR is this? (check all applicable)

## Description
This PR is beginning the process of porting over the search results page to PostgreSQL FTS from Elasticsearch. Our game plan is to convert each endpoint one by one. This means the controller is going to look quite gnarly during this transition period, please bear with us 😅 

This PR adds the comment search behind a feature flag. Once the feature flag is enabled, any search with the comments filter will use PostgreSQL instead of Elasticsearch.

_**Please note there are code and PR comments below explaining more.**_

## Related Tickets & Documents
RFC 0153: https://github.com/forem/rfcs/pull/153
https://github.com/orgs/forem/projects/29#card-58641161

## QA Instructions, Screenshots, Recordings
1. Log into an account (local admin or your own).
2. Leave a comment on an article.
3. Go to search (http://localhost:3000/search).
4. Type a search term in the search box at the top related to the comment you left in step 2. Then click the `Comments` filter on the left.
5. Verify your comment shows up in the search results.
6. Fire up a console and activate the feature flag (`FeatureFlag.enable(:search_2_comments)`).
7. Repeat steps 3 - 5 and make sure your comment is included in the results :tada:.

## Added tests?

## [Forem core team only] How will this change be communicated?

## [optional] Are there any post deployment tasks we need to perform?
I'll coordinate with the systems team to activate the feature flag.


![dog_typing_on_computer](https://media.giphy.com/media/wpoLqr5FT1sY0/giphy.gif)
